src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -563,24 +554,13 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
                     // Otherwise the value is another object (perhaps an HtmlString) and we'll ask it to format itself.
                     string stringValue;
 
-                    // Intentionally using is+cast here for performance reasons. This is more performant than as+bool?
-                    // because of boxing.
-                    if (val.Value is bool)
+                    if (!ShouldRenderAttributeValue(val))
```

Comment:
If existing tests are passing, it's to the implicit conversion from `PositionTagged<object>` to `object` or because existing tests don't cover `null` and `false` cases at all. Be explicit and avoid the repeated / buried conversion regardless.

Separately extend `RazorPageTest` to exercise `WriteAttribute[To]()` more thoroughly.